app/src/main/java/org/fossasia/openevent/app/event/attendees/AttendeesPresenter.java

```diff
@@ -0,0 +1,68 @@
+package org.fossasia.openevent.app.event.attendees;
+
+import android.content.Context;
+
+import org.fossasia.openevent.app.data.EventDataRepository;
+import org.fossasia.openevent.app.data.UtilModel;
+import org.fossasia.openevent.app.data.contract.IEventDataRepository;
+import org.fossasia.openevent.app.data.contract.IUtilModel;
+import org.fossasia.openevent.app.data.models.Attendee;
+import org.fossasia.openevent.app.event.attendees.contract.IAttendeesPresenter;
+import org.fossasia.openevent.app.event.attendees.contract.IAttendeesView;
+
+public class AttendeesPresenter implements IAttendeesPresenter {
+
+    IEventDataRepository eventDataRepository;
+    IAttendeesView attendeesView;
+    Long eventId;
+
+    public AttendeesPresenter(IAttendeesView attendeesView, Context context, Long eventId) {
+        this.attendeesView = attendeesView;
+        eventDataRepository = new EventDataRepository(new UtilModel(context));
+        this.eventId = eventId;
+    }
+
+    @Override
+    public void attach() {
+        loadAttendees(true);
```

Comment:
The essence of MVP is never to delegate a behavior of one class to another class. If we want to define the behavior of model such that it updates the result every time we fetch, then we will define it in model and presenter will just call for data and it will return the cached result and then update the view when the view when data is available automatically, but till it is required, let's take it literally, load from cache if it's saved, and use network only if it's not. Since we are using in-memory cache, the results will never persist across sessions, so we never force user to see too stale data.

So, *when* the model is coded to be smart enough, it will be triggered to give us cached result and the model itself will fetch the result and save it to refresh the view later. 

Read about offline first apps, where we act locally and sync globally. We want to make the experience as smooth as possible, but we can't jump to conclusions about what users need. We must assume they don't want to fetch the data every second and if they are offline, the current code won't show the previous data. So, we must always at least show the cached data, doesn't matter if we fetch 10000 results or not